### PR TITLE
Proper reloading of learning rate scheduler state at training resumption

### PR DESCRIPTION
**Pull request by fhieber**
#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by fhieber** [comment]
This should fix #366 .

**Comment by fhieber** [comment]
`Trainer.initialize_optimizer()` is just a wrapper for `self.model.initialize_optimizer`, but this is not related to the bug. The bug was that `OptimizerConfig` had two references to the `lr_scheduler` instance, one as a member of `OptimizerConfig`, and once in the optimizer_params (which are used by MXNet). At training resumption we unpickled the lr_scheduler instance, but only replaced `OptimizerConfig.lr_scheduler` with that pickle object, not the optimizer_params for MXNet.
This change makes sure there is only one reference held.
In what line would you like a comment?
